mmseg/models/decode_heads/decode_head.py

```diff
@@ -84,6 +91,12 @@ def __init__(self,
         self.ignore_index = ignore_index
         self.align_corners = align_corners
 
+        if self.num_classes == 2 and self.out_channels is None:
+            warnings.warn('set out_channels to 1 to reduce the number of model\
+                parameters')
+        elif self.num_classes == 2 and self.out_channels == 1:
+            self.num_classes = 1
```

Comment:
if self. out_channels is None:
    self.out_channels = self.num_classes
    if self.num_classes == 2:
        warning.warn('For binary segmentation, we suggest to use `out_channels = 1` to define the output channels of segmentor', and use `threshold` to  convert seg_logist into a prediction applying a threshold